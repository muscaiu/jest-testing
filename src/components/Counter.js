import React, { Component } from 'react'

class Counter extends Component {
    state = {
        count: 0
    }

    makeIncrementer = amount => () => {
        this.setState(prevState => ({
            count: prevState.count + amount
        }))
    }

    increment = this.makeIncrementer(1)
    decrement = this.makeIncrementer(-1)

    render() {
        return (
            <div>
                <p>Current count: {this.state.count}</p>
                <button
                    className="increment"
                    onClick={this.increment}
                >
                    Increment count
                </button>
                <button
                    className="decrement"
                    onClick={this.decrement}
                >
                    Increment count
                </button>
            </div>
        )
    }
}

export default Counter
