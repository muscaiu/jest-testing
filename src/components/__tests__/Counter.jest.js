import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Counter from '../Counter'

const mock = jest.fn();
mock.mockReturnValue(false);

describe('Counter component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Counter />).toJSON()
    
    expect (tree).toMatchSnapshot()
  })
  it('starts with a count of 0', () => {
    const wrapper = shallow(<Counter />)
    const text = wrapper.find('p').text()

    expect(text).toEqual('Current count: 0')
  })

  it('increments the count when the button is clicked', () => {
    const wrapper = shallow(<Counter />)
    const incrementBtn = wrapper.find('button.increment')
    incrementBtn.simulate('click')
    const text = wrapper.find('p').text()

    expect(text).toEqual('Current count: 1')
  })

  it('decrements the count when the decrement button is clicked', () => {
    const wrapper = shallow(<Counter />)
    const decrementBtn = wrapper.find('button.decrement')
    decrementBtn.simulate('click')
    const text = wrapper.find('p').text()

    expect(text).toEqual('Current count: -1')
  })
})
