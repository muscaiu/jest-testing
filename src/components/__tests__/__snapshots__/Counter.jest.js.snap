// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component matches the snapshot 1`] = `
<div>
  <p>
    Current count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    +
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
